### DISCUSSION TITLE
Array into metadata of vector

### DISCUSSION THREAD

**Discussion by MUSS-Davs** [neon-sunset/Pinecone.NET#79] [Q&A] [answered]
Hi, in the example the code is like this:

`var vectors = new[] 
{ 
  new Vector 
    {
      Id = "vector1",
      Values = new float[] { 0.1f, 0.2f, 0.3f }, 
      Metadata = new MetadataMap 
      {
         ["genre"] = "horror", 
         ["duration"] = 120
      }
    }
};
`

Now suppose I have a list of identifiers

`List<string> idList = new() { "id1", "id2", "id3" };`

How can i add this to the metadata?, I've try this: 

`var vectors = new[]
{
    new Vector
    {
        Id = "vector1",
        Values = new float[] { 0.1f, 0.2f, 0.3f },
        Metadata = new MetadataMap
        {
            ["genre"] = "horror",
            ["duration"] = 120,
            ["idList"] = idList 
        }
    }
};`

but it doesn't work.
I need this for search with filters like: $in or $nin

**Comment by neon-sunset** [comment, marked as answer]
Hi @MUSS-Davs ,
This is a restriction due to how metadata operates in relation to C# being strongly typed. Because `MetadataMap` is a dictionary of `string:MetadataValue`s, `List<string>` has to be converted to `List<MetadataValue>` which itself can be implicitly converted to `MetadataValue` node.

The way to deal with this is to map sequences / maps of your values to corresponding `MetadataValue`s like this:
```csharp
var idList = new List<string> { "id1", "id2", "id3" };

var metadata = new MetadataMap
{
    ["genre"] = "horror",
    ["duration"] = 120,
    ["idList"] = idList
        .Select(id => (MetadataValue)id)
        .ToArray()
};

var vector = new Vector
{
    Id = "vector1",
    Values = [0.1f, 0.2f, 0.3f],
    Metadata = metadata
};
```

**Reply by neon-sunset** [reply]
Since this is far from the first time the issue with metadata is raised, I've released an new version which should help with this: https://github.com/neon-sunset/Pinecone.NET/releases/tag/1.3.2
